File: Ceaser.fs
let shiftE (letter, move) =
    let mutable ascii = ((int(letter) - 65 + move)%26)
    if ascii < 0 then
        ascii <- ascii + 26
    char(ascii + 65)
    
let shiftD (letter, move) =
    let mutable ascii = ((int(letter) - 65 - move)%26)
    if ascii < 0 then
        ascii <- ascii + 26
    char(ascii + 65)

    
let Encrypt (word : string) (move : int) : string =
    printfn "%s" "This is the Encryption: "
    let shifted = String.map (fun i -> shiftE(i, move)) word
    printfn "%s" shifted
    shifted
        
let Decrypt (word : string) (move : int) =
    printfn "%s" "This is the Decryption: "
    let shifted = String.map (fun i -> shiftD(i, move)) word
    printfn "%s" shifted
    
let rec Solve (word : string) (move : int) =
    let shifted = String.map (fun i -> shiftE(i, move)) word
    printfn "%s" shifted
    if move < 26 then
        let mutable i = move
        i <- i + 1
        Solve "SORINZ" i

let mutable word = "SORINZ"
let move = 5
word <- Encrypt word move
printfn "%s" "This is Solve: "
Solve "SORINZ" 1
Decrypt word move

File: Ceaser.erl
-module(prog).
-export([main/0]).

shift1(Char, Move) ->
	Ascii = (hd(Char) - $A + Move + 26) rem (26),
	if
		Ascii < 0 ->
			Ascii = Ascii + 26;
		true ->
			Ascii
	end,
	Ascii + $A.

shift2(Char, Move) ->
	Ascii = (hd(Char) - $A - Move + 26) rem (26),
	if
		Ascii < 0 ->
			Ascii = Ascii + 26;
		true ->
			Ascii
	end,
	Ascii + $A.
	
encrypt(Test, Move) ->
    Encrypt = lists:map(fun(Char) -> shift1([Char], Move) end, Test),
    EncryptString = lists:flatten(Encrypt),
    io:fwrite("~s~n", [EncryptString]),
    EncryptString.
	
decrypt(Test, Move) ->
	Decrypt = lists:map(fun(Char) -> shift2([Char], Move) end, Test),
    io:fwrite("~p~n", [Decrypt]).
	
solve(Test, Move, Place) ->
	Solve = lists:map(fun(Char) -> shift1([Char], Move) end, Test),
	io:fwrite("~p~n",[Solve]),
	if
		Place < 25 ->
			solve("SORINZ", Move+1, Place+1);
		true ->
			true
	end.
	
main() ->
	Test = "SORINZ",
	Move = 5,
	io:fwrite("This is Encrypt:\n"),
	Test2 = encrypt(Test, Move),
	io:fwrite("This is Solve:\n"),
	solve(Test, 1, 0),
	io:fwrite("This is Decrypt:\n"),
	decrypt(Test2, Move).
